ios/FluentUI/Navigation/UINavigationItem+Navigation.swift

```diff
@@ -31,6 +33,28 @@ import UIKit
         }
     }
 
+    /// An wide accessory view that can be shown as a subview of ShyHeaderView but doesn't have leading, trailing
+    /// and bottom insets. So it can appear as being part of the content view but still contract and expand as part of
+    /// the shy header.
+    var wideAccessoryView: UIView? {
+        get {
+            return objc_getAssociatedObject(self, &AssociatedKeys.wideAccessoryView) as? UIView
+        }
+        set {
+            objc_setAssociatedObject(self, &AssociatedKeys.wideAccessoryView, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
+        }
+    }
+
+    /// The height of the wideAccessoryView.
+    var wideAccessoryViewHeight: CGFloat {
```

Comment:
Can't we just use the height of the view instead of asking for it to be explicitly provided in the navitem?

Reply:
My understanding is that we can't. Because not all subviews of `UIView` have `intrinsicContentSize`, so there is no reliable way to get the size from the provided view. And before this change the `maxHeight` is 36.0(the height of the SearchBar) + top and bottom insets, and I guess the reason is also because of that.

Reply:
Curious what other people think. I would love to know if we can do that.

Reply:
the height of searchbar is fixed so I think it just worked out conveniently that we could calculate `maxHeight` based off the height of the search bar and insets. however, I'm also curious if we can just get the height from the view. does using `systemLayoutSizeFitting` work?

Reply:
`systemLayoutSizeFitting ` does work! I didn't know we can do that! Thank you so much to you both!